src/compiler/program.ts

```diff
@@ -329,6 +329,7 @@ namespace ts {
         let files: SourceFile[] = [];
         let fileProcessingDiagnostics = createDiagnosticCollection();
         let programDiagnostics = createDiagnosticCollection();
+        let emitBlockingDiagnostics = createDiagnosticCollection();

```

Comment:
they do not change. at least not meant to. files and options should be treated as immutable.